### PR TITLE
Install provider via Crossplane as a package in local dev environment

### PR DESCRIPTION
**Pull request by ccremer**
## Summary

* This PR gets rid of the Helm chart and installs the provider as a Crossplane package instead
* Seeing how Crossplane packages work and their installation, a local registry is required to avoid having to push a temporary image to a public registry just for testing.

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
  `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
  as they show up in the changelog
- [x] Link this PR to related issues

### CODE REVIEW

**Comment by ccremer** [comment]
For testing this PR: 
Follow the quickstart section in the readme and inspect the results

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Building the package fails for me with a linting error:
```
gmake package-provider
kubectl crossplane: error: failed to build package: failed to lint package: not exactly one package meta type
gmake: *** [package/package.mk:20: package-provider] Error 1
```

EDIT: renaming `crossplane.yaml.template` to `crossplane.yaml` fixes the issue. I guess something is supposed to transform the template file?

**Comment by ZuGao** [comment]
Having the same issue as @Kidswiss:
`kubectl crossplane: error: failed to build package: failed to lint package: not exactly one package meta type`

File: operator/bucketcontroller/setup.go [hunk 1]
```diff
@@ -11,13 +11,6 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
-// +kubebuilder:rbac:groups=cloudscale.crossplane.io,resources=buckets,verbs=get;list;watch;create;update;patch;delete
```
[the remark this answers is not in the document]
Comment (ccremer) [reply operator/bucketcontroller/setup.go:14]: It's magic by Crossplane installation. First I was under the impression I had to add the rules to access the CRDs and put them under `permissionRequests`, turns out this caused more harm than help. My guess is that Crossplane analyzes the CRDs and gives basically enough permissions to the SA.
Now that I have removed these markers and also removed the yq command that imported the kubebuilder-generated RBAC rules before to the permissionRequests you noticed the issue with `crossplane.yaml.template`. Oversight from my side, I'll fix it.

**Comment by ccremer** [comment]
I've fixed the package generation. you're invited to test again

**Review by Kidswiss — APPROVED** [review]
Seems to be working now.

**Pull request merged**